File: 2020/3.py
import sys

def parse(f):
    return [line.strip() for line in f]

trees = parse(sys.stdin)
print(trees)
x, y = 0, 0
slope = 3, 1
n = 0
while y < len(trees):
    if trees[y][x % len(trees[y])] == '#':
        n += 1
    x, y = x + slope[0], y + slope[1]
print(n)
        

File: 2020/3.rs
use std::io::{self, BufRead};
fn main() {
    let stdin = io::stdin();
    let trees: Vec<_> = stdin.lock().lines().map(|line| line.unwrap()).collect();
    let mut x: usize = 0;
    let mut y: usize = 0;
    let dx: usize = 3;
    let dy: usize = 1;
    let mut n: i32 = 0;
    while y < trees.len() {
        if trees[y].chars().nth(x % trees[y].len()) == Some('#') {
            n += 1
        }
        x += dx;
        y += dy;
    }
    println!("{}", n);
}